lib/services/ngsi/entities-NGSI-v2.js

```diff
@@ -489,176 +375,580 @@ function sendUpdateValueNgsi2(entityName, attributes, typeInformation, token, ca
+                for (let n = 0; n < payload.entities.length; n++) {
+                    // include metadata with TimeInstant in attrs when TimeInstant is provided as measure in all entities
+                    payload.entities[n] = addTimestampNgsi2(
+                        payload.entities[n],
+                        typeInformation.timezone,
+                        timestampValue,
+                        false // skipMetadataAtt
                     );
-                    callback(null);
                 }
+            } else {
+                // Do not include metadata with TimeInstant in attrs when TimeInstant is provided as measure
+                // and no more entities
+                payload.entities[0] = addTimestampNgsi2(
+                    payload.entities[0],
+                    typeInformation.timezone,
+                    timestampValue,
+                    true // skipMetadataAtt
+                );
+            }
+        } else {
+            // jshint maxdepth:5
+            for (let n = 0; n < payload.entities.length; n++) {
+                if (!utils.isTimestampedNgsi2(payload.entities[n])) {
+                    // legacy check needed?
+                    payload.entities[n] = addTimestampNgsi2(payload.entities[n], typeInformation.timezone);
+                    // jshint maxdepth:5
+                } else if (!utils.IsValidTimestampedNgsi2(payload.entities[n])) {
+                    // legacy check needed?
+                    logger.error(context, 'Invalid timestamp:%s', JSON.stringify(payload.entities[0]));
+                    callback(new errors.BadTimestamp(payload.entities));
+                    return;
+                }
+            }
+        }
+    }
+    logger.debug(context, 'sendUpdateValueNgsi2Refactored \n ending payload=%j', payload);
+
+    for (let m = 0; m < payload.entities.length; m++) {
+        for (var key in payload.entities[m]) {
+            // purge object_id from payload
+            if (payload.entities[m][key] && payload.entities[m][key].object_id) {
+                delete payload.entities[m][key].object_id;
             }
         }
+        payload.entities[m] = NGSIUtils.castJsonNativeAttributes(payload.entities[m]); // native types
+    }
+    logger.debug(
+        context,
+        'sendUpdateValueNgsi2Refactored \n payload with native types and without object_id %j',
+        payload
     );
+
+    options.json = payload;
+
+    // Prevent to update an entity with an empty payload
+    if (
+        Object.keys(options.json).length > 0 &&
+        (options.json.entities.length > 1 ||
+            (options.json.entities.length === 1 && Object.keys(options.json.entities[0]).length > 2)) // more than id and type
+    ) {
+        // Final check: (to keep tests unchanged) before do CB requests
+        //              one entity     -> request /v2/entities/ + entityName + /atts ?type=typeInformation.type
+        //              multi entities -> request v2/op/update
```

Comment:
Should be explained (with a comment) that in that case we don't do anything as the current options object has been already prepared for this case (or something like that).